### PR TITLE
fix flaky test card tests

### PR DESCRIPTION
**Pull request by fzhao99**
# FRONTEND PULL REQUEST

Noticed there was [a](https://github.com/CDCgov/prime-simplereport/actions/runs/9473750072/job/26101996426) [flaky](https://github.com/CDCgov/prime-simplereport/actions/runs/9469747069/job/26089159521) [test](https://github.com/CDCgov/prime-simplereport/actions/runs/9469051367/job/26086853339) that we added after some of the recent test card additions, so putting up a PR to fix. 

Took the opportunity to do some fixture refactoring as well since adding in the mock was way more of a pain in the neck due to the needed constants being split between files.

Think the core issue here is that our current setup requiring interaction-to-interaction testing mocks _seems_ to work sometimes with Apollo mocking provider when missing, but then fails other times. The larger fix is to use a system that can just reuse mocks rather than having to manually specify it every time, which we might be able to do with our work with msw.

## Changes Proposed
PR is mainly moving a bunch of test constants around so the diff is bigger than it looks. Should be a no-op that can be tested by re-running the testing job

- Adds a missing mock to solve the flaking issue
- Centralized a bunch of similar mocks / constants / fixtures into one central file for the testing queue
- The Sonar duplication flag is a result of needing to copy / past test fixtures so I've ignored it but if it's an issue for anyone, happy to fix

## Testing
Rerun the frontend testing process a few times and verify they pass determinantly. Got a string of successful runs in a row working [here](https://github.com/CDCgov/prime-simplereport/actions/runs/9471165250) (red X failure is from Sonar duplication issues, even though the actual frontend test job is working fine)

### CODE REVIEW

**Review by bobbywells52 — CHANGES_REQUESTED** [review]
I've ran the whole suite 5 times now locally on this branch and only gotten a pass on the first go for 3 of the runs. The failed tests resolve after one re-run but I'm definitely not getting a determinate run of tests.

**Comment by fzhao99** [comment]
Bobby found some other flaking tests due to a similar issue so trying to fix them. If anyone finds any other ones please let me know!

File: frontend/src/app/testQueue/TestCard/TestCard.test.tsx [hunk 1]
```diff
@@ -114,214 +107,167 @@ async function getSpecimenTypeDropdown() {
...
-
-  const testOrderInfo: QueriedTestOrder = {
-    internalId: "1b02363b-ce71-4f30-a2d6-d82b56a91b39",
-    dateAdded: "2022-11-08 13:33:07.503",
-    symptoms:
-      '{"64531003":"false","103001002":"false","84229001":"false","68235000":"false","426000000":"false","49727002":"false","68962001":"false","422587007":"false","267036007":"false","62315008":"false","43724002":"false","36955009":"false","44169009":"false","422400008":"false","230145002":"false","25064002":"false","162397003":"false"}',
-    symptomOnset: null,
-    noSymptoms: true,
-    deviceType: {
+const facilityInfo: QueriedFacility = {
```
Comment (mehansen) [inline frontend/src/app/testQueue/TestCard/TestCard.test.tsx:110]: is this `facilityInfo` a duplicate of the [`facilityInfo`](https://github.com/CDCgov/prime-simplereport/pull/7792/files#diff-eb68b11555006268ed4f5c502b29cf4d76f8453e34e522b9fd99884c1f249f8bR296) defined in `testCardTestConstants.ts`?

Comment (fzhao99) [reply frontend/src/app/testQueue/TestCard/TestCard.test.tsx:110]: yep, good catch! for some reason earlier I was having cross-file import issues for this variable where importing it from the constants file it used to live in was causing errors, but it seems like it's working fine now 🤷

**Comment by fzhao99** [comment]
@bobbywells52 I think I've gotten these to pass pretty consistently locally / [got a string of 5 successful runs in CI](https://github.com/CDCgov/prime-simplereport/actions/runs/9501990961). Guessing even if we didn't get to the root cause of the flakiness that this will help, so going to propose merging it.

Happy to pair further on whether there might be another cause / config misalignment locally

**Comment by bobbywells52** [comment]
> @bobbywells52 I've gotten these to pass pretty consistently locally / [got a string of successful runs in CI](https://github.com/CDCgov/prime-simplereport/actions/runs/9501990961). Guessing even if we didn't get to the root cause of the flakiness that this will help, so going to propose merging it.
> 
> Happy to pair further on whether there might be another cause / config misalignment locally

Taking a look rn -- if I get a good string of runs I'll approve. Def down to pair and get to the root!

**Review by bobbywells52 — APPROVED** [review]
Everything is determinately passing for me this morning -- lgtm!

**Review by emyl3 — APPROVED** [review]
LGTM! Thanks for fixing this Bob!

**Review by mpbrown — APPROVED** [review]
LGTM locally and in github actions, thanks for investigating and fixing this Bob!

**Pull request merged**